packages/payload/src/admin/components/elements/DocumentControls/index.tsx

```diff
@@ -58,6 +58,8 @@ export const DocumentControls: React.FC<{
 
   const hasCreatePermission = 'create' in permissions && permissions.create?.permission
   const hasDeletePermission = 'delete' in permissions && permissions.delete?.permission
+  const hasPublishPermission =
+    'publish' in permissions && permissions.fields?._status?.update?.permission
```

Comment:
I know this is a WIP, just wanted to note that `publish` does not exist on the `permissions` type as of right now.